pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -568,7 +568,7 @@ def check_ipython_compatibility(self):
         if self._distrib_type in DistributedType.ipython_compatible_types():
             return
         # check ipython env
-        if "IPython" in sys.modules:
+        if _module_available("IPython"):
```

Comment:
The failing tests seem to be related to this line. Any ideas?

Reply:
Any idea why `IPython` does not define `__spec__` @Borda?

Should we keep `"IPython" in sys.modules` or update `_module_available`?

Reply:
Any suggestions? I think if we keep `"IPython" in sys.modules`, this PR is ready to be merged.

Reply:
Fixed it. You forgot to remove

`@mock.patch.dict(sys.modules, {"IPython": Mock()})`

Reply:
Thanks for the patch and sorry for my sloppiness. But I see even after your commit, there are still failing test cases on `test_accelerator_connector.py::test_ipython_incompatible_backend_error` and there's not so much useful information provided in the log. Are the failures related to `_module_available()` or to `get_ipython()`?

Reply:
Maybe we should use this to check if it's interactive

https://stackoverflow.com/a/64523765

Instead of relying on IPython

Reply:
Done in https://github.com/PyTorchLightning/pytorch-lightning/pull/5970/commits/990c03ef6ecafb4c3a82a85bb54ac89c6d3bfcb4

Reply:
Seems reasonable. Thanks!